### ISSUE TITLE
Infer generated columns

### ISSUE THREAD

**Issue by RobinBlomberg** [RobinBlomberg/kysely-codegen#2]
For this SQL table (using auto-increment):
```sql
CREATE TABLE my_table(
  id SERIAL
);
```
Instead of this:
```typescript
export interface MyTable {
  id: number;
}
```
We should generate this:
```typescript
export interface MyTable {
  id: Generated<number>;
}
```

---

This ensures that you can insert new rows in Kysely without having to provide generated fields.

It is possible that this may require some adjustment of the Kysely introspector.

**Comment by koskimas** [comment]
The [version I just released](https://github.com/koskimas/kysely/releases/tag/0.19.6) has `isAutoIncrementing` and `hasDefaultValue` properties in `ColumnMetadata`. You can use those the add `Generated` fields.

**Comment by RobinBlomberg** [comment]
Wow, amazing! Nice work—I'll try to make time to add this functionality later today.

**Comment by RobinBlomberg** [comment]
Published in 0.3.0! 🌟
